collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/InvalidFormTest.kt

```diff
@@ -0,0 +1,63 @@
+package org.odk.collect.android.feature.formentry
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import org.odk.collect.android.support.pages.FormEntryPage
+import org.odk.collect.android.support.pages.MainMenuPage
+import org.odk.collect.android.support.rules.CollectTestRule
+import org.odk.collect.android.support.rules.TestRuleChain.chain
+import org.odk.collect.strings.R
+
+@RunWith(AndroidJUnit4::class)
+class InvalidFormTest {
+    private var rule = CollectTestRule()
+
+    @get:Rule
+    var copyFormChain: RuleChain = chain().around(rule)
+
+    @Test
+    fun brokenForm_shouldNotBeVisibleOnFormList() {
+        rule.startAtMainMenu()
+            .copyForm("invalid-form.xml")
+            .clickFillBlankForm()
+            .checkIsSnackbarErrorVisible("org.javarosa.xform.parse.XFormParseException: Cycle detected in form's relevant and calculation logic!")
+            .assertTextDoesNotExist("invalid-form")
+    }
+
+    @Test
+    fun app_ShouldNotCrash_whenFillingFormsWithErrors() {
```

Comment:
I think we should break this into a test per form so we can describe the error type more clearly.

Reply:
Done in https://github.com/getodk/collect/pull/6295/commits/efe2c60b9cefd2bcf1b270c827ce44b7a987f4f5. I think we can just keep two of those forms.